src/code/PreRender.c

```diff
@@ -254,195 +351,385 @@ void func_80170AE0(PreRender* this, Gfx** gfxp, s32 alpha) {
-                for (j = 1; j < 15; j += 2) {
-                    if ((i != j) && (buffG[j] >= buffG[i]) && (buffA[j] == 7)) {
-                        pxG = buffG[i];
+            if (pmaxG < buffG[i]) {
+                for (j = 1; j < 5 * 3; j += 2) {
+                    if ((i != j) && (buffG[j] >= buffG[i]) && (buffCvg[j] == 7)) {
+                        pmaxG = buffG[i];
                     }
                 }
             }
-            if (pxB < buffB[i]) {
-                for (j = 1; j < 15; j += 2) {
-                    if ((i != j) && (buffB[j] >= buffB[i]) && (buffA[j] == 7)) {
-                        pxB = buffB[i];
+            if (pmaxB < buffB[i]) {
+                for (j = 1; j < 5 * 3; j += 2) {
+                    if ((i != j) && (buffB[j] >= buffB[i]) && (buffCvg[j] == 7)) {
+                        pmaxB = buffB[i];
                     }
                 }
             }
+
             if (1) {}
-            if (pxR2 > buffR[i]) {
-                for (j = 1; j < 15; j += 2) {
-                    if ((i != j) && (buffR[j] <= buffR[i]) && (buffA[j] == 7)) {
-                        pxR2 = buffR[i];
+
+            // Determine "Penultimate Minimum" Value
+
+            // Same as above with inverted conditions
+            if (pminR > buffR[i]) {
+                for (j = 1; j < 5 * 3; j += 2) {
+                    if ((i != j) && (buffR[j] <= buffR[i]) && (buffCvg[j] == 7)) {
+                        pminR = buffR[i];
                     }
                 }
             }
-            if (pxG2 > buffG[i]) {
-                for (j = 1; j < 15; j += 2) {
-                    if ((i != j) && (buffG[j] <= buffG[i]) && (buffA[j] == 7)) {
-                        pxG2 = buffG[i];
+            if (pminG > buffG[i]) {
+                for (j = 1; j < 5 * 3; j += 2) {
+                    if ((i != j) && (buffG[j] <= buffG[i]) && (buffCvg[j] == 7)) {
+                        pminG = buffG[i];
                     }
                 }
             }
-            if (pxB2 > buffB[i]) {
-                for (j = 1; j < 15; j += 2) {
-                    if ((i != j) && (buffB[j] <= buffB[i]) && (buffA[j] == 7)) {
-                        pxB2 = buffB[i];
+            if (pminB > buffB[i]) {
+                for (j = 1; j < 5 * 3; j += 2) {
+                    if ((i != j) && (buffB[j] <= buffB[i]) && (buffCvg[j] == 7)) {
+                        pminB = buffB[i];
                     }
                 }
             }
         }
     }
 
-    pad = 7 - buffA[7];
-    pxR3 = buffR[7] + (((s32)((pad * ((pxR + pxR2) - (buffR[7] << 1))) + 4)) >> 3);
-    pxG3 = buffG[7] + (((s32)((pad * ((pxG + pxG2) - (buffG[7] << 1))) + 4)) >> 3);
-    pxB3 = buffB[7] + (((s32)((pad * ((pxB + pxB2) - (buffB[7] << 1))) + 4)) >> 3);
+    // The background color is determined by averaging the penultimate minimum and maximum pixels, and subtracting the
+    // ForeGround color:
+    //      BackGround = (pMax + pMin) - (ForeGround) * 2
```

Comment:
Any particular reason why you use `background` (lowercase "g") in the first line, then `ForeGround` and `BackGround` (capital "g") in the following lines?